.github/workflows/approved-for-ci-run.yml

```diff
@@ -88,15 +99,50 @@ jobs:
             Feel free to review/comment/discuss the original PR #${PR_NUMBER}.
           EOF
 
-          ALREADY_CREATED="$(gh pr --repo ${GITHUB_REPOSITORY} list --head ${BRANCH} --base main --json number --jq '.[].number')"
+          ALREADY_CREATED=${{ steps.get_pr.outputs.ALREADY_CREATED }}
           if [ -z "${ALREADY_CREATED}" ]; then
+            LABELS=$(gh pr view ${PR_NUMBER}  --json labels --jq '.labels.[].name' | \
+            awk 'BEGIN {labels="run-e2e-tests-in-draft"} {if ($1 ~ /^run/) labels=labels "," $1;} END {print labels}')
             gh pr --repo "${GITHUB_REPOSITORY}" create --title "CI run for PR #${PR_NUMBER}" \
                                                        --body-file "body.md" \
                                                        --head "${BRANCH}" \
                                                        --base "main" \
-                                                       --label "run-e2e-tests-in-draft" \
+                                                       --label $LABELS \
                                                        --draft
+          else
+            LABELS_TO_REMOVE=$(comm -23 <(gh pr --repo ${GITHUB_REPOSITORY} view $ALREADY_CREATED --json labels --jq '.labels.[].name'| grep -E '^run' | sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+            LABELS_TO_REMOVE=${LABELS_TO_REMOVE//run-e2e-tests-in-draft/}
+            LABELS_TO_REMOVE=$(echo $LABELS_TO_REMOVE | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+            LABELS_TO_ADD=$(comm -13 <(gh pr --repo ${GITHUB_REPOSITORY} view $ALREADY_CREATED --json labels --jq '.labels.[].name'|  grep -E '^run' |sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+            LABELS_TO_ADD=$(echo $LABELS_TO_ADD | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+            echo "LABELS_TO_ADD=$LABELS_TO_ADD" >> $GITHUB_OUTPUT
+            echo "LABELS_TO_REMOVE=$LABELS_TO_REMOVE" >> $GITHUB_OUTPUT
+          fi
+      - name: Modify the existant pull request (if required)
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ALREADY_CREATED=${{ steps.get_pr.outputs.ALREADY_CREATED }}
+          if [ -z "${ALREADY_CREATED}" ] ; then 
+            exit 0
+          fi
```

Comment:
We can move this if to a higher level, i.e. `if: steps.get_pr.outputs.ALREADY_CREATED != ''`.
Also, it can incorporate this logic:
```
          if [ -z "${LABELS_TO_ADD}" ] && [ -z "${LABELS_TO_REMOVE}" ] ; then 
            exit 0
          fi
```

Reply:
done